File: agent/library/src/main/java/org/matilda/messages/handlers/MessageDispatcher.java
package org.matilda.messages.handlers;

import org.matilda.messages.Message;

import java.util.concurrent.ExecutorService;

public class MessageDispatcher implements MessageHandler {
    private final ExecutorService mExecutorService;
    private final MessageHandler mMessageHandler;

    public MessageDispatcher(ExecutorService executorService, MessageHandler messageHandler) {
        mExecutorService = executorService;
        mMessageHandler = messageHandler;
    }

    @Override
    public void handle(Message message) {
        mExecutorService.submit(() -> mMessageHandler.handle(message));
    }
}

File: agent/library/src/main/java/org/matilda/messages/handlers/MessageDispatcher.kt
package org.matilda.messages.handlers

import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch
import org.matilda.messages.Message

class MessageDispatcher(private val mCoroutineScope: CoroutineScope, private val mMessageHandler: MessageHandler) :
    MessageHandler {
    override suspend fun handle(message: Message) {
        mCoroutineScope.launch { mMessageHandler.handle(message) }
    }
}
